### ISSUE TITLE
Video doesn't work in release mode on iOS with text tracks (6.0.0-alpha.3)

### ISSUE THREAD

**Issue by sam1463** [TheWidlarzGroup/react-native-video#2890]
# Bug

## Platform

Which player are you experiencing the problem on:
* iOS

## Environment info

React native info output:

```bash
 // paste it here
System:
    OS: macOS 12.2.1
    CPU: (10) x64 Apple M1 Pro
    Memory: 29.35 MB / 16.00 GB
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 14.18.3 - ~/.nvm/versions/node/v14.18.3/bin/node
    Yarn: 1.22.18 - ~/.nvm/versions/node/v14.15.5/bin/yarn
    npm: 6.14.15 - ~/.nvm/versions/node/v14.18.3/bin/npm
    Watchman: 2022.10.03.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK:
      API Levels: 30, 31, 32, 33
      Build Tools: 30.0.2, 30.0.3, 31.0.0, 32.0.0, 32.1.0
      System Images: android-30 | ARM 64 v8a, android-30 | Intel x86 Atom_64, android-30 | Google APIs ARM 64 v8a, android-30 | Google APIs Intel x86 Atom, android-30 | Google APIs Intel x86 Atom_64, android-30 | Google Play ARM 64 v8a, android-31 | ARM 64 v8a, android-31 | Google APIs ARM 64 v8a, android-31 | Google Play ARM 64 v8a, android-32 | Google APIs ARM 64 v8a, android-32 | Google Play ARM 64 v8a, android-33 | Google APIs ARM 64 v8a, android-Tiramisu | Google APIs ARM 64 v8a, android-Tiramisu | Google Play ARM 64 v8a
      Android NDK: Not Found
  IDEs:
    Android Studio: 2021.1 AI-211.7628.21.2111.8309675
    Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
  Languages:
    Java: 11.0.14.1 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.0.0 => 18.0.0 
    react-native: 0.69.5 => 0.69.5 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

Library version: 6.0.0-alpha.3

## Steps To Reproduce

1. use react-native-video to render a video with `textTracks` supplied with a remote URI and a `selectedTextTrack`
2. create an iOS build in release mode, and you get a black screen with a red X and "Video unavailable" text.
...

## Expected behaviour

1. The video should play with subtitles.
2. It's broken in release builds on simulator and physical device, but it works correctly in debug builds on iOS (sim + real device) and both debug + release schemes work on android

## Video sample
If possible, include a link to the video that has the problem that can be streamed or downloaded from.

**Comment by freeboub** [comment]
Looks like similar to https://github.com/react-native-video/react-native-video/issues/2811

**Comment by cihanbas** [comment]
same issue

**Comment by drdosan** [comment]
Same problem, anyone with a solution? It will be if it wouldn't be something permission in info.plist?

**Comment by drdosan** [comment]
Guys, I managed to solve by reducing the version to version 5.2.1! Maybe this can help some of you!

**Comment by sam1463** [comment]
Yeah sorry if this issue wasn't clear, but this is only a problem on the v6 alpha. However, since the v6 alpha is now the main branch of this repo, switching to 5.2.1 means losing out on many recent changes, such as support for captions

**Comment by dvijeniii05** [comment]
Still an issue for v6 alpha.4.

**Comment by dvijeniii05** [comment]
We found our issue and it was related to building app through XCode 13. Building though XCode 14+ resolved the issue for us :')

**Comment by clementnv** [comment]
Hello,
I have the same issue on alpha 6.

When I have `textTracks`, crash on iOS only, with Xcode 14.

```
react_native_video/RCTVideoUtils.swift:190: Fatal error: Unexpectedly found nil while implicitly unwrapping an Optional value
2023-08-09 22:37:43.831568+0200 KephyreTraining[3667:319229] react_native_video/RCTVideoUtils.swift:190: Fatal error: Unexpectedly found nil while implicitly unwrapping an Optional value
```

**Comment by freeboub** [comment]
There is no easy way to reproduce the issue in ticket description.
Linked tickets has been fixed for 6.0.0-alpha.11 or beta.1
I will close the issue let me know if we need to reopen it.
